src/archive/src/main.rs

```diff
@@ -0,0 +1,394 @@
+use candid::{CandidType, Deserialize, Principal};
+use ic_cdk::api::stable::stable64_size;
+use ic_cdk::api::time;
+use ic_cdk::{caller, trap};
+use ic_cdk_macros::{init, post_upgrade, query, update};
+use ic_stable_structures::memory_manager::{MemoryId, MemoryManager, VirtualMemory};
+use ic_stable_structures::{
+    cell::Cell as StableCell, log::Log, DefaultMemoryImpl, Memory as StableMemory,
+    RestrictedMemory, StableBTreeMap, Storable,
+};
+use internet_identity_interface::*;
+use metrics_encoder::MetricsEncoder;
+use serde_bytes::ByteBuf;
+use std::borrow::Cow;
+use std::cell::RefCell;
+
+type Memory = RestrictedMemory<DefaultMemoryImpl>;
+type StableLog = Log<VirtualMemory<Memory>, VirtualMemory<Memory>>;
+type ConfigCell = StableCell<ArchiveConfig, Memory>;
+type UserIndex = StableBTreeMap<VirtualMemory<Memory>, UserIndexKey, ()>;
```

Comment:
I've been squinting at the index code for a while and still don't feel confident about my understanding of how it works/what it does.

What does it do? How does it achieve this? Why are all values of the BTree empty?

Reply:
I added a comment explaining this as a top level module doc.